extensions-contrib/opentelemetry-extensions/src/test/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReaderTest.java

```diff
@@ -358,6 +358,29 @@ public void testInvalidProtobuf() {
     Assert.assertThrows(ParseException.class, () -> rows.hasNext());
     Assert.assertThrows(ParseException.class, () -> rows.next());
   }
+  
+  @Test
+  public void testInvalidMetricType() {
+    metricBuilder
+        .setName("deprecated_intsum")
+        .getIntSumBuilder()
+        .addDataPointsBuilder()
+        .setTimeUnixNano(TIMESTAMP);
+
+    MetricsData metricsData = metricsDataBuilder.build();
+
+    CloseableIterator<InputRow> rows = new OpenTelemetryMetricsProtobufReader(
+        dimensionsSpec,
+        new ByteEntity(metricsData.toByteArray()),
+        "metric.name",
+        "raw.value",
+        "descriptor.",
+        "custom."
+    ).read();
+
+    Assert.assertThrows(ParseException.class, () -> rows.hasNext());
```

Comment:
have we confirmed this behavior is what we want? Will throwing an exception on hasNext cause Druid to ignore the entire batch of metrics? Ideally we'd want to ignore rows with unsupported metric types but still consider those we do support.

Reply:
Right. I feel the IllegalStateException is just unnecessary here, it makes more sense to just return an empty list if the metric type is invalid